Editor/Utility/CinemachineSceneTools.cs

```diff
@@ -0,0 +1,564 @@
+            Handles.color = trackedObjectOffsetHandleIsUsedOrHovered ? 
+                Handles.selectedColor : CinemachineSettings.CinemachineCoreSettings.ActiveGizmoColour;
+            Handles.DrawDottedLine(lookAtPos, trackedObjectPos, k_DottedLineSpacing);
+            Handles.DrawLine(trackedObjectPos, cmComponent.VcamState.FinalPosition);
+            Handles.color = originalColor;
+
+            SoloOnDrag(
+                trackedObjectOffsetHandleIsDragged, cmComponent.VirtualCamera, tooHandleMaxId, ref soloSetByTools);
+        }
+
+        public static void TransposerFollowOffsetTool(CinemachineTransposer cmComponent, ref bool soloSetByTools)
+        {
+            var brain = CinemachineCore.Instance.FindPotentialTargetBrain(cmComponent.VirtualCamera);
+            var up = brain != null ? brain.DefaultWorldUp : Vector3.up;
+            var camPos = cmComponent.GetTargetCameraPosition(up);
+            var camRot = cmComponent.GetReferenceOrientation(up);
+
+            EditorGUI.BeginChangeCheck();
+            var foHandleMinId = GUIUtility.GetControlID(FocusType.Passive); // TODO: KGB workaround until id is exposed
+            var newPos = Handles.PositionHandle(camPos, camRot);
+            var foHandleMaxId = GUIUtility.GetControlID(FocusType.Passive); // TODO: KGB workaround until id is exposed
+            if (EditorGUI.EndChangeCheck())
+            {
+                Undo.RecordObject(cmComponent, 
+                    "Change Follow Offset Position using handle in Scene View.");
+                
+                cmComponent.m_FollowOffset += PositionHandleDelta(camRot, newPos, camPos);
+                cmComponent.m_FollowOffset = cmComponent.EffectiveOffset; // sanitize offset
+                
+                InspectorUtility.RepaintGameView();
+            }
+
+            var followOffsetHandleIsDragged = 
+                foHandleMinId < GUIUtility.hotControl && GUIUtility.hotControl < foHandleMaxId;
+            var followOffsetHandleIsDraggedOrHovered = followOffsetHandleIsDragged || 
+                foHandleMinId < HandleUtility.nearestControl && HandleUtility.nearestControl < foHandleMaxId;
+            if (followOffsetHandleIsDraggedOrHovered)
+            {
+                DrawLabel(camPos, "Follow offset " + cmComponent.m_FollowOffset.ToString("F1"));
+            }
+            var originalColor = Handles.color;
+            Handles.color = followOffsetHandleIsDraggedOrHovered ? 
+                Handles.selectedColor : CinemachineSettings.CinemachineCoreSettings.ActiveGizmoColour;
+            Handles.DrawDottedLine(cmComponent.FollowTargetPosition, camPos, k_DottedLineSpacing);
+            Handles.color = originalColor;
+
+            SoloOnDrag(followOffsetHandleIsDragged, cmComponent.VirtualCamera, foHandleMaxId, ref soloSetByTools);
+        }
+        
+        public static void SoloOnDrag(
+            bool isDragged, CinemachineVirtualCameraBase vcam, int handleMaxId, ref bool soloSetByTools)
```

Comment:
Why are you have a soloSetByTools param?  Can't you just have a local static and manage it yourself?  It would simplify the callers.